### PR TITLE
Separate OpenAPIController routes and rendering

### PR DESCRIPTION
**Pull request by mobiusxs**
Issue #356.

This PR splits `redoc` and the base path/root handlers. The `root` handler serves the original `redoc` content, but enables easier overriding of the rendering and routing on the `OpenAPIController`.

- separate UI rendering and routing
- add routed `root` method
- separate testing of `/` and `redoc`
-  add `root` override example

### CODE REVIEW

**Comment by Goldziher** [comment]
Currently failing validation. Please run pre-commit (see CONTRIBUTING.md)

File: docs/usage/12-openapi.md [hunk 1]
```diff
@@ -163,3 +163,17 @@ app = Starlite(
     openapi_config=OpenAPIConfig(openapi_controller=MyOpenAPIController),
 )
 ```
+
+You can override the `root` handler to serve other included UIs or your own [template](./15-templating.md):
```
Comment (Goldziher) [inline docs/usage/12-openapi.md:167]: Please explain about the different render methods available.

Also -what if users want to have both swaggerUI and redoc?

File: starlite/openapi/controller.py [hunk 2]
```diff
@@ -34,6 +34,11 @@ def schema_from_request(request: Request) -> "OpenAPI":
             raise ImproperlyConfiguredException("Starlite has not been instantiated with OpenAPIConfig")
         return request.app.openapi_schema
 
+    @get(path="/", media_type=MediaType.HTML, include_in_schema=False)
+    def root(self, request: Request) -> str:
+        """Endpoint that serves root content"""
```
Comment (Goldziher) [inline starlite/openapi/controller.py:39]: this comment doesn't explain what root content is. Please explain

File: docs/usage/12-openapi.md (hunk 1 above)
Comment (mobiusxs) [reply docs/usage/12-openapi.md:167]: How do you feel about this: 

> The root handler serves Redoc by default via `render_redoc`. You can override the `root` handler to serve other included documentation, such as Swagger UI by returning `render_swagger_ui` or Stoplight Elements with `render_stoplight_elements`. You can also have it serve your own [template](./15-templating.md):

Do you think it should include the part about rendering templates? I'm not sure if it's relevant, but just wanted to highlight the flexibility.

Comment (Goldziher) [reply docs/usage/12-openapi.md:167]: Sure thats fine

File: starlite/openapi/controller.py (hunk 2 above)
Comment (mobiusxs) [reply starlite/openapi/controller.py:39]: Refactored it to this:

```
@get(path="/", media_type=MediaType.HTML, include_in_schema=False)
def root(self, request: Request) -> str:
    """
    This route handler serves Redoc API documentation at the root path.
    Override this handler to serve custom templates or other API documentation, such as
    `render_swagger_ui`, `render_stoplight_elements`, a template, or your own content.
    """
    return self.render_redoc(request)
```
What do you think?

Comment (Goldziher) [reply starlite/openapi/controller.py]: Looks good

**Comment by Goldziher** [comment]
@all-contributors add @mobiusxs code

**Pull request merged**